src/onnx/parse_shrink.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_shrink : op_parser<parse_shrink>
+{
+    std::vector<op_desc> operators() const { return {{"Shrink"}}; }
+
+    instruction_ref parse(const op_desc&,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        float bias = 0.0;
+        if(contains(info.attributes, "bias"))
+        {
+            bias = parser.parse_value(info.attributes.at("bias")).at<float>();
+        }
+        float lambd = 0.5;
+        if(contains(info.attributes, "lambd"))
+        {
+            lambd = parser.parse_value(info.attributes.at("lambd")).at<float>();
+        }
+
+        auto x_shape = args[0]->get_shape();
+        auto lit_bias =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {bias}});
+        auto lit_neg_lambd =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {-lambd}});
+        auto lit_lambd =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {lambd}});
+        auto lit_zero = info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {0}});
+
+        instruction_ref mb_bias;
+        instruction_ref mb_neg_lambd;
+        instruction_ref mb_lambd;
+        instruction_ref mb_zero;
+        if(x_shape.dynamic())
```

Comment:
You can use `add_common_op` instead. 
That would handle broadcasting and conversion of data type for you. 
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/e5cd8b6b0ec30f7a3b7b6775b096c3a41003b09d/src/onnx/onnx_parser.cpp#L172

Reply:
bias and lambda is only of dtype `float` but input can have dtype other than float. Therefore need to handle conversions as well. Can you add tests for those as well ?

Reply:
I've added onnx test cases for these.